freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaScalingValidationService.java

```diff
@@ -46,13 +43,26 @@ public void validateStackForVerticalUpscale(Stack stack, VerticalScaleRequest re
     }
 
     public void validateStackForDownscale(Set<InstanceMetaData> allInstances, Stack stack, ScalingPath scalingPath) {
-        if (allInstances.size() < AvailabilityType.HA.getInstanceCount()) {
-            LOGGER.warn("FreeIPA instance count is not allowed. Size: [{}]", allInstances.size());
-            throw new BadRequestException("Downscaling currently only available for FreeIPA installation with 3 instances");
-        }
+        validateScalingIsDownscale(scalingPath);
         executeCommonValidations(allInstances, stack, scalingPath, OperationType.DOWNSCALE);
     }
 
+    private void validateScalingIsUpscale(ScalingPath scalingPath) {
+        if (scalingPath.getOriginalAvailabilityType().getInstanceCount() > scalingPath.getTargetAvailabilityType().getInstanceCount()) {
```

Comment:
shouldn't we compare with the actual original instance count here?

Reply:
There are 2 separate validations.
This validation only checks that the availability type is not going to decrease (i.e. the user sent a downscale request to the upscale endpoint). So an HA => HA scaling is also allowed.
There is another validation (in the common part) where the instance count is checked if it will change at all. If not, a validation error is thrown.

Reply:
makes sense, thank you